### PR TITLE
Add flag to get attention from fusion

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Currently GPT gets the attention_mask from the fused softmax kernel.

If, the kernel isn't used, then the fallback to pytorch softmax breaks.

This PR makes it configurable to get attention_mask from the kernel.

**Collection**: NLP

# Changelog 
- Update forward and loss func

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by MaximumEntropy — COMMENTED** [review]
This looks good to me. @erhoo82 to confirm if you are okay with the default value of the new arg as False.

**Review by MaximumEntropy — APPROVED** [review]
